### DISCUSSION TITLE
Djockey, a Djot-based documentation system and static site generator

### DISCUSSION THREAD

**Discussion by irskep** [jgm/djot#314] [Show and tell]
tl;dr [Djockey's docs are here](https://steveasleep.com/djockey/), and you can invoke it locally with `npx djockey` or `bunx jockey`, with a single-binary-not-requiring-a-Node-runtime release coming in the future.

I've been thinking about what my "perfect" documentation system would look like for a few years and was always hung up on the markup language. reStructuredText is extensible but painful and tied to Python, and Markdown has no syntax for extensibility and the plugin ecosystem is a mess of hacks.

Then I discovered Djot, which solves all my markup language problems. So I went ahead and built that "perfect" documentation system.

Some design goals:
- Make Djot a serious choice for writing project documentation for anything
- Provide an on-ramp and off-ramp for Markdown users, including "GitHub Markdown docs checked into git" as an _output_ target
- Give the JS ecosystem a proper tree-based documentation tool
- Give the Djot community more data about use cases ([Markdown interop](https://github.com/jgm/djot/discussions/219), [extensibility](https://github.com/jgm/djot/discussions/77), and a place to implement multi-document features like [inclusion](https://github.com/jgm/djot/issues/307) and [cross-references](https://github.com/jgm/djot/issues/30) without making the core parser more complex

There are a hundred more micro-opinions in the design, but those are the big ones.

Djockey will be in an experimental phase until further notice. I would love feedback, especially if you have a project but there's something blocking you from adopting Djockey for your docs. Besides being experimental, of course. :-)

**Comment by jgm** [comment]
I added a link to the README, under Tooling.

**Comment by ryanabx** [comment]
This is amazing! I made a [simple static-site generator](https://github.com/ryanabx/djot-ssg) for my own use cases a week ago, but I might consider moving to something like this which is more full-featured. Excellent work

**Reply by irskep** [reply]
Thanks so much! Feel free to make some feature requests, since you'd be my first user. :-)
